rest/src/main/java/fi/otavanopisto/pyramus/rest/controller/StudentController.java

```diff
@@ -537,6 +542,33 @@ public void deleteLodgingPeriod(StudentLodgingPeriod period) {
 
     return result;
   }
+  
+  /* Checks if logged user is guidance counselor of specific student */
+  public boolean amIGuidanceCouncelor(Long studentId, StaffMember staffMember) {
+    
+    Student student = findStudentById(studentId);
+    // List of student's user groups
+    List<StudentGroup> studentGroups = studentGroupDAO.listByStudent(student, 0, 50, false);
```

Comment:
Why not a `StudentGroupDAO` method something like `listByStudentAndUserAndGuidanceGroupAndArchived` which would return all student groups that...

a) have been marked as guidance groups
b) have the student in them
c) have the staff member in them
d) are not archived

Return value of this method would then simply be whether the DAO method returned an empty list or not.